_posts/francois_gerthoffert/2016-12-09-Adding-Support-For-Multiple-Authors-in-Jekyll.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: post
+title:  "Adding support for multiple authors in Jekyll"
+breadcrumb: true
+author: francois_gerthoffert
+date: 2016-12-08
+categories: francois_gerthoffert
+tags:
+    - Blog
+    - Jekyll
+    - Github
+teaser:
+    info: OICR being a research institute, a portion of the team, in particular those of us more on the science side, is used to Academic Publishing and its related conventions such as Academic Authorship. It was not long before we were asked for supporting multiple authors in a blog post.
+    image: francois_gerthoffert/jekyll-logo.png
+
+---
+
+OICR being a research institute, a portion of the team, in particular those of us more on the science side, is used to [Academic Publishing](https://en.wikipedia.org/wiki/Academic_publishing) and its related conventions such as [Academic Authorship](https://en.wikipedia.org/wiki/Academic_authorship). It was not long before we were asked for supporting multiple authors in a blog post.
+
+This blog post will detail the solution we implemented to add this feature into our Jekyll implementation of the blog hosted on [GitHub Pages](https://pages.github.com/).
+
+## Jekyll & Liquid
+
+The blog is built on [Jekyll](http://jekyllrb.com/), which uses [Liquid](https://shopify.github.io/liquid/) for its template engine. Liquid provides basic operators, types and a set of filters to implement logic into templates.
+
+## Declaring Authors in a blog post
+
+There was multiple options to declare authors in a blog post, I was initially looking at having different types for the 'author' variable, and then depending of the type, implement a different logic to display authors.
+
+~~~
+# One Author:
+author: francois_gerthoffert
+
+# Multiple Authors
+author:
+   - francois_gerthoffert
+   - jared_baker
+~~~
+
+Sadly Liquid did not allow type checking, so our solution used a simpler approach, using different variable names when having one or having multiple authors
+
+~~~
+# One Author:
+author: francois_gerthoffert
+
+# Multiple Authors
+authors:
+   - francois_gerthoffert
+   - jared_baker
+~~~
+
+## Listing Authors
+
+To list authors, the system will first count the size of the "author" and "authors" variable declared in the blog post, returning 0 if the variable does not exists or is empty.
+
+~~~
```

Comment:
Didn't look further than adding the code section and seeing {% being interpreted by jekyll